benchmark/utils_sd.js

```diff
@@ -0,0 +1,204 @@
+      image.crossOrigin = '';
+      image.onload = () => {
+        canvas.width = width;
+        canvas.height = height;
+        canvas.setAttribute("width", width);
+        canvas.setAttribute("height", height);
+        ctx.drawImage(image, 0, 0, width, height);
+        resolve(image);
+      };
+    });
+    image.src = imageElement.src;
+    return promise;
+  }
+
+  async executeAsync() {
+    let computeResults = [];
+    let decodeResults = [];
+    let singlePose = null;
+    for (let i = 0; i < this.iterations; i++) {
+      this.onExecuteSingle(i);
+      await new Promise(resolve => requestAnimationFrame(resolve));
+      let tStart = performance.now();
+      await this.executeSingleAsync();
+      let elapsedTime = performance.now() - tStart;
+      computeResults.push(elapsedTime);
+      let dstart = performance.now();
+      singlePose = decodeSinglepose(sigmoid(this.heatmapTensor),
+                                    this.offsetTensor,
+                                    toHeatmapsize(this.scaleInputSize,
+                                                  this.outputStride),
+                                    this.outputStride);
+      let decodeTime = performance.now() - dstart;
+      console.log("Decode time:" + decodeTime);
+      decodeResults.push(decodeTime);
+    }
+    // draw canvas by last result
+    await this.loadImage(showCanvasElement, imageElement.width, imageElement.height);
+    let ctx = showCanvasElement.getContext('2d');
+    let scaleX = showCanvasElement.width / this.scaleWidth;
+    let scaleY = showCanvasElement.height / this.scaleHeight;
+    singlePose.forEach((pose) => {
+      if (pose.score >= this.minScore) {
+        drawKeypoints(pose.keypoints, this.minScore, ctx, scaleX, scaleY);
+        drawSkeleton(pose.keypoints, this.minScore, ctx, scaleX, scaleY);
+      }
+    });
+    bkImageSrc = imageElement.src;
+    imageElement.src = showCanvasElement.toDataURL();
+    return {
+      "computeResults": computeResults,
+      "decodeResults": decodeResults
+    };
+  }
+
+  handleResults(results) {
+    let profilingResults = null;
+    if (this.backend !== 'WebNN') {
+      profilingResults = this.model._compilation._preparedModel.dumpProfilingResults();
+    }
+    return {
+      "computeResults": summarize(results.computeResults),
+      "decodeResults": summarize(results.decodeResults),
+      "profilingResults": summarizeProf(profilingResults)
+    };
+  }
+
+  /**
+   * Finalize
+   * @returns {Promise<void>}
+   */
+  finalize() {
+    super.finalize();
+    this.modelInfoDict = null;
+    this.model = null;
+    this.labels = null;
+    this.inputTensor = null;
+    this.inputSize = null;
+    this.modelVersion = null;
+    this.outputStride = null;
+    this.scaleFactor = null;
+    this.minScore = null;
+    this.scaleWidth = null;
+    this.scaleHeight = null;
+    this.scaleInputSize = null;
+    this.heatmapTensor = null;
+    this.offsetTensor  = null;
+  }
```

Comment:
```
this.modelInfoDict = null;
    this.model = null;
    this.labels = null;
    this.inputTensor = null;
    this.inputSize = null;
    this.modelVersion = null;
    this.outputStride = null;
    this.scaleFactor = null;
    this.minScore = null;
    this.scaleWidth = null;
    this.scaleHeight = null;
    this.scaleInputSize = null;
    this.heatmapTensor = null;
    this.offsetTensor  = null;
```
Please reconsider if this kind of code block are necessary, also please invoke deleteAll().

```
  deleteAll() {
    if (this.model._backend != 'WebML') {
      this.model._compilation._preparedModel._deleteAll();
    }
  }
```